cmd/warpforge/main_test.go

```diff
@@ -48,19 +67,30 @@ func testFile(t *testing.T, fileName string, workDir *string) {
 
 	doc.BuildDirIndex()
 	patches := testmark.PatchAccumulator{}
+outer:
 	for _, dir := range doc.DirEnt.ChildrenList {
 		testName := dir.Name
+		tags := newTagSet()
 		testDir := dir
-		if _, hasNetTests := dir.Children["net"]; hasNetTests {
-			if *testutil.FlagOffline {
-				t.Log("skipping test", dir.Name, "due to offline flag")
-				continue
+		for _, x := range dir.ChildrenList {
+			//iterating on children just to find out if it has tags sucks but it's the only way
```

Comment:
seems a bit much to say this sucks :P

"iterating on data necessary in order to respond to it", oh no :P

Reply:
Well we went from O(1) to O(n) on that operation so my jimmies are rustled.

Reply:
I cleaned this up a bit. Still O(n) but at least it's not using labels.